src/Juvix/Compiler/Internal/Translation/FromInternal/Analysis/TypeChecking/Data/Inference.hs

```diff
@@ -213,7 +261,7 @@ weakNormalize' = go
     goApp (Application l r i) = do
       l' <- go l
       case l' of
-        ExpressionSimpleLambda (SimpleLambda lamVar _ lamBody) -> do
+        ExpressionSimpleLambda (SimpleLambda (SimpleBinder lamVar _) lamBody) -> do
           go (substitutionE (HashMap.singleton lamVar r) lamBody)
```

Comment:
As above. I think this is not entirely correct.

Reply:
BTW, I can't find any place in the pipeline where `SimpleLambda` is created. Isn't this dead code which should be removed?

Reply:
> BTW, I can't find any place in the pipeline where `SimpleLambda` is created. Isn't this dead code which should be removed?

It is used in `Inference.functionDefEval`. 
`SimpleLambda` represents the subset of lambdas that we can normalize during inference (basically type synonyms).